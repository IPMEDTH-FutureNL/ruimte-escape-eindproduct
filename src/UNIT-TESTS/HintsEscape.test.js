import { render , screen, fireEvent } from "@testing-library/react"
import {BrowserRouter as Router} from 'react-router-dom';
import HintEscape from "../components/HintsEscape";

test("check if hints escape get rendered on start up", () => {
    const { asFragment } = render(
        <Router>
            <HintEscape></HintEscape>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})

test("check if hints escape get rendered for clickinggame", () => {
    const { asFragment } = render(
        <Router>
            <HintEscape type="clickHints"></HintEscape>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})  

test("check if hints escape get rendered for drag and drop", () => {
    const { asFragment } = render(
        <Router>
            <HintEscape type="dragAndDropHints"></HintEscape>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})

test("check if hints escape get rendered for oplosmatrix", () => {
    const { asFragment } = render(
        <Router>
            <HintEscape type="oplosmatrixHints"></HintEscape>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})

test("check if hints escape get rendered for rebus", () => {
    const { asFragment } = render(
        <Router>
            <HintEscape type="rebusHints"></HintEscape>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})

test("check if hints escape get rendered for quiz", () => {
    const { asFragment } = render(
        <Router>
            <HintEscape type="quizzHints"></HintEscape>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})