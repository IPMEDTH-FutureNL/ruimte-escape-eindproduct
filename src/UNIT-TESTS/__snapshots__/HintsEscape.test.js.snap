// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if hints escape get rendered for clickinggame 1`] = `
<DocumentFragment>
  <button
    aria-describedby="popup-2"
    class="hint-button"
  >
     Tips
  </button>
  <div
    class="punten"
  >
    Punten: 
  </div>
</DocumentFragment>
`;

exports[`check if hints escape get rendered for drag and drop 1`] = `
<DocumentFragment>
  <button
    aria-describedby="popup-4"
    class="hint-button"
  >
     Tips
  </button>
  <div
    class="punten"
  >
    Punten: 
  </div>
</DocumentFragment>
`;

exports[`check if hints escape get rendered for oplosmatrix 1`] = `
<DocumentFragment>
  <button
    aria-describedby="popup-6"
    class="hint-button"
  >
     Tips
  </button>
  <div
    class="punten"
  >
    Punten: 
  </div>
</DocumentFragment>
`;

exports[`check if hints escape get rendered for quiz 1`] = `
<DocumentFragment>
  <button
    aria-describedby="popup-10"
    class="hint-button"
  >
     Tips
  </button>
  <div
    class="punten"
  >
    Punten: 
  </div>
</DocumentFragment>
`;

exports[`check if hints escape get rendered for rebus 1`] = `
<DocumentFragment>
  <button
    aria-describedby="popup-8"
    class="hint-button"
  >
     Tips
  </button>
  <div
    class="punten"
  >
    Punten: 
  </div>
</DocumentFragment>
`;

exports[`check if hints escape get rendered on start up 1`] = `
<DocumentFragment>
  <button
    aria-describedby="popup-1"
    class="hint-button"
  >
     Tips
  </button>
  <div
    class="punten"
  >
    Punten: 
  </div>
</DocumentFragment>
`;
